File: app/src/main/java/sgtmelon/scriptum/element/ColorDialog.java
package sgtmelon.scriptum.element;

import android.app.Dialog;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.SimpleItemAnimator;
import sgtmelon.scriptum.R;
import sgtmelon.scriptum.app.adapter.ColorAdapter;
import sgtmelon.scriptum.office.annot.def.DialogDef;
import sgtmelon.scriptum.office.blank.DialogBlank;
import sgtmelon.scriptum.office.intf.ItemIntf;

public final class ColorDialog extends DialogBlank implements ItemIntf.Click {

    private int init, check;

    private int[] icons, colors;

    public void setArguments(int check) {
        Bundle arg = new Bundle();

        arg.putInt(DialogDef.INIT, check);
        arg.putInt(DialogDef.VALUE, check);

        setArguments(arg);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Bundle arg = getArguments();

        if (savedInstanceState != null) {
            init = savedInstanceState.getInt(DialogDef.INIT);
            check = savedInstanceState.getInt(DialogDef.VALUE);
        } else if (arg != null) {
            init = arg.getInt(DialogDef.INIT);
            check = arg.getInt(DialogDef.VALUE);
        }

        RecyclerView recyclerView = new RecyclerView(context);

        int padding = context.getResources().getInteger(R.integer.dlg_recycler_padding);
        recyclerView.setPadding(padding, padding, padding, padding);
        recyclerView.setOverScrollMode(View.OVER_SCROLL_NEVER);

        GridLayoutManager layoutManager = new GridLayoutManager(context, context.getResources().getInteger(R.integer.dlg_color_column));
        recyclerView.setLayoutManager(layoutManager);

        ColorAdapter adapter = new ColorAdapter(context, check, icons, colors);
        adapter.setClick(this);
        recyclerView.setAdapter(adapter);

        SimpleItemAnimator animator = (SimpleItemAnimator) recyclerView.getItemAnimator();
        if (animator != null) animator.setSupportsChangeAnimations(false);

        return new AlertDialog.Builder(context)
                .setTitle(title)
                .setView(recyclerView)
                .setPositiveButton(getString(R.string.dialog_btn_accept), onPositiveClick)
                .setNegativeButton(getString(R.string.dialog_btn_cancel), (dialog, id) -> dialog.cancel())
                .setCancelable(true)
                .create();
    }

    public void setIcons(int[] icons) {
        this.icons = icons;
    }

    public void setColors(int[] colors) {
        this.colors = colors;
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);

        outState.putInt(DialogDef.INIT, init);
        outState.putInt(DialogDef.VALUE, check);
    }

    public int getCheck() {
        return check;
    }

    @Override
    protected void setEnable() {
        super.setEnable();

        if (init == check) buttonPositive.setEnabled(false);
        else buttonPositive.setEnabled(true);
    }

    @Override
    public void onItemClick(View view, int p) {
        check = p;
        setEnable();
    }
}

File: app/src/main/java/sgtmelon/scriptum/widget/color/ColorDialog.kt
package sgtmelon.scriptum.widget.color

import android.app.Dialog
import android.os.Bundle
import android.view.View

import androidx.appcompat.app.AlertDialog
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.SimpleItemAnimator
import sgtmelon.safedialog.DialogBlank
import sgtmelon.scriptum.R

class ColorDialog : DialogBlank(), ColorIntf.ClickListener {

    private var init: Int = 0

    var check: Int = 0
        private set

    fun setArguments(check: Int) {
        val bundle = Bundle()

        bundle.putInt(INIT, check)
        bundle.putInt(VALUE, check)

        arguments = bundle
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val bundle = arguments

        init = savedInstanceState?.getInt(INIT)
                ?: bundle?.getInt(INIT)
                ?: 0

        check = savedInstanceState?.getInt(VALUE)
                ?: bundle?.getInt(VALUE)
                ?: 0

        val adapter = ColorAdapter(activity)
        adapter.setCheck(check)
        adapter.setClickListener(this)

        val padding = 24

        val recyclerView = RecyclerView(activity)
        recyclerView.setPadding(padding, padding, padding, padding)
        recyclerView.overScrollMode = View.OVER_SCROLL_NEVER
        recyclerView.layoutManager = GridLayoutManager(activity, activity.resources.getInteger(R.integer.column_color))
        recyclerView.adapter = adapter
        (recyclerView.itemAnimator as SimpleItemAnimator).supportsChangeAnimations = false

        return AlertDialog.Builder(activity)
                .setTitle(title)
                .setView(recyclerView)
                .setPositiveButton(getString(R.string.dialog_btn_accept), onPositiveClick)
                .setNegativeButton(getString(R.string.dialog_btn_cancel)) { dialog, _ -> dialog.cancel() }
                .setCancelable(true)
                .create()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)

        outState.putInt(INIT, init)
        outState.putInt(VALUE, check)
    }

    override fun setEnable() {
        super.setEnable()
        buttonPositive.isEnabled = init != check
    }

    override fun onColorClick(p: Int) {
        check = p
        setEnable()
    }

}